packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx

```diff
@@ -10,24 +10,53 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { ReactElement, useEffect } from "react";
+import { ReactElement, FC, useEffect } from "react";
 
 import { SupportedLanguages } from "./OdysseyTranslationProvider.types";
 
-import i18n from "./OdysseyI18n";
+import i18n, { defaultNS, resources } from "./OdysseyI18n";
 import { I18nextProvider } from "react-i18next";
 
-export function OdysseyTranslationProvider({
-  children,
-  languageCode,
-}: {
+export type TranslationOverrides = {
+  [key in SupportedLanguages]?: Partial<typeof resources["en"]>;
+};
+
+interface OdysseyTranslationProviderProps {
   children: ReactElement;
   languageCode?: SupportedLanguages;
-}) {
-  useEffect(() => {
-    // Defaults to the browser's language if available otherwise `en` will be used
-    i18n.changeLanguage(languageCode || window.navigator.language);
-  }, [languageCode]);
-
-  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
+  translationOverrides?: TranslationOverrides;
 }
+
+const mergeBundleOverrides = (
+  languageCode: SupportedLanguages,
+  translationOverrides: TranslationOverrides
+) => {
+  const bundle = resources[languageCode];
+  const overrides = translationOverrides[languageCode];
+  return {
+    ...bundle,
+    ...overrides,
+  };
+};
+
+export const OdysseyTranslationProvider: FC<OdysseyTranslationProviderProps> =
+  ({ children, languageCode, translationOverrides }) => {
+    useEffect(() => {
+      // Defaults to the browser's language if available otherwise `en` will be used
+      i18n.changeLanguage(languageCode || window.navigator.language);
+    }, [languageCode]);
+
+    useEffect(() => {
+      if (translationOverrides) {
+        Object.keys(translationOverrides).forEach((lng) => {
+          const bundle = mergeBundleOverrides(
+            lng as SupportedLanguages,
+            translationOverrides
+          );
+          i18n.addResourceBundle(lng, defaultNS, bundle);
+        });
```

Comment:
I'd prefer we make this more verbose:

```suggestion
        Object.keys(translationOverrides).forEach((language) => {
          const bundle = mergeBundleOverrides(
            language as SupportedLanguages,
            translationOverrides
          );
          i18n.addResourceBundle(language, defaultNS, bundle);
        });
```

Reply:
Right, that is possible as we renamed the prop to `languageCode` ;)